05 11 textpreprocessor

henryonai opened the pull request:
# Content

## Add Tokenization and Normalization Utilities

This PR introduces two classes, `Tokenize` and `Normalization`, which provide essential text processing utilities using NLTK.

### Tokenize
- Implements methods for tokenizing text:
  - **whitespace_tokenizer()**: Tokenizes text by whitespace.
  - **word_punct_tokenizer()**: Separates words and punctuation.
  - **treebank_word_tokenizer()**: Tokenizes using Penn Treebank rules.

### Normalization
- Provides basic normalization methods:
  - **stemming()**: Reduces tokens to their root form using PorterStemmer.
  - **lemmatization()**: Converts tokens to their base form using WordNetLemmatizer.

henryonai committed:
feat(tokenize): implement basic text tokenization methods

Add a Tokenize class that provides methods for tokenizing text using WhitespaceTokenizer, WordPunctTokenizer, and TreebankWordTokenizer from NLTK. Also includes an example usage in the main script to demonstrate functionality.

- whitespace_tokenizer(): Tokenizes text by whitespace.
- word_punct_tokenizer(): Tokenizes text by separating words and punctuation.
- treebank_word_tokenizer(): Tokenizes text using Penn Treebank rules.

henryonai committed:
feat(normalization): add text normalization class with stemming and lemmatization

Introduce the Normalization class to provide basic text normalization methods.
This class includes stemming using PorterStemmer and lemmatization using WordNetLemmatizer from NLTK.

- stemming(): Reduces each token to its root form using PorterStemmer.
- lemmatization(): Converts each token to its base or dictionary form using WordNetLemmatizer.

ngluonghungg2611 commented on 05_11_textpreprocessor/models/tokenization.py:
```diff
@@ -0,0 +1,90 @@
+        List[str]
+            A list of tokens separated by whitespace.
+        """
+        tokenizer = WhitespaceTokenizer()
```
khởi tạo cái này ở init em nhé

ngluonghungg2611 commented on 05_11_textpreprocessor/models/tokenization.py:
```diff
@@ -0,0 +1,90 @@
+        List[str]
+            A list of tokens, with punctuation marks as separate tokens.
+        """
+        tokenizer = WordPunctTokenizer()
```
khởi tạo cái này ở init em nhé

ngluonghungg2611 commented on 05_11_textpreprocessor/models/tokenization.py:
```diff
@@ -0,0 +1,90 @@
+        List[str]
+            A list of tokens based on Treebank tokenization rules.
+        """
+        tokenizer = TreebankWordTokenizer()
```
khởi tạo cái này ở init em nhé

henryonai committed:
refactor(tokenize):Refactor Tokenize class to enhance tokenization functionality

Added tokenizer attributes (WhitespaceTokenizer, WordPunctTokenizer, TreebankWordTokenizer) as instance variables for cleaner code.
